modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/internal/search/spi/model/index/contributor/MBMessageModelIndexerWriterContributor.java

```diff
@@ -0,0 +1,149 @@
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.message.boards.internal.search.spi.model.index.contributor;
+
+import com.liferay.document.library.kernel.model.DLFileEntry;
+import com.liferay.message.boards.model.MBMessage;
+import com.liferay.message.boards.service.MBMessageLocalService;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.dao.orm.Property;
+import com.liferay.portal.kernel.dao.orm.PropertyFactoryUtil;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.exception.SystemException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.repository.model.FileEntry;
+import com.liferay.portal.kernel.search.Document;
+import com.liferay.portal.kernel.search.Indexer;
+import com.liferay.portal.kernel.search.IndexerRegistryUtil;
+import com.liferay.portal.kernel.search.SearchException;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+import com.liferay.portal.search.batch.BatchIndexingActionable;
+import com.liferay.portal.search.batch.DynamicQueryBatchIndexingActionableFactory;
+import com.liferay.portal.search.spi.model.index.contributor.ModelIndexerWriterContributor;
+import com.liferay.portal.search.spi.model.index.contributor.helper.IndexerWriterMode;
+import com.liferay.portal.search.spi.model.index.contributor.helper.ModelIndexerWriterDocumentHelper;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Luan Maoski
+ */
+@Component(
+	immediate = true,
+	property = "indexer.class.name=com.liferay.message.boards.model.MBMessage",
+	service = ModelIndexerWriterContributor.class
+)
+public class MBMessageModelIndexerWriterContributor
+	implements ModelIndexerWriterContributor<MBMessage> {
+
+	@Override
+	public void customize(
+		BatchIndexingActionable batchIndexingActionable,
+		ModelIndexerWriterDocumentHelper modelIndexerWriterDocumentHelper) {
+
+		batchIndexingActionable.setAddCriteriaMethod(
+			dynamicQuery -> {
+				Property statusProperty = PropertyFactoryUtil.forName("status");
+
+				Integer[] statuses = {
+					WorkflowConstants.STATUS_APPROVED,
+					WorkflowConstants.STATUS_IN_TRASH
+				};
+
+				dynamicQuery.add(statusProperty.in(statuses));
+
+			});
+
+		batchIndexingActionable.setPerformActionMethod(
+			(MBMessage mbMessage) -> {
+				if (_log.isDebugEnabled()) {
```

Comment:
Hi @brandizzi  , the three `MB` tests are also failing before [migration](https://github.com/luanmaoski/liferay-portal/pull/19).

For some reason the `_log` is disable. Also, when i run the source format locally it get stuck at 69%.